### DISCUSSION TITLE
[@types/jest] Is the type definitions for Jest need any longer

### DISCUSSION THREAD

**Discussion by SayakMukhopadhyay** [DefinitelyTyped/DefinitelyTyped#62736] [Issues with a @types package]
I have noticed that Jest is providing its own types and even mentioning in its [README](https://github.com/facebook/jest/blob/main/packages/jest-types/README.md) that the type definitions in here *might* not be up to date, which to me is an indication that they are going to maintain the types in their projects.

As such, is there any need to maintain this definition any longer? I had seen a [discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/57277#issuecomment-975728724) almost a year back where it was mentioned that Jest recommends Definitely Typed. I don't see such recommendations any longer ( the only recommendation is to use this library if devs don't want to import the type definitions, which I don't find a very appealing argument).

[a comment by an automated account is not shown]

**Comment by mrazauskas** [comment]
Jest documentation tries to warn that `@types/jest` do not update their types automatically with each release. Typings must be added manually to `@types/jest` and there will be some lag. In contrary Jest always ships accurate and up to date types which get updated each time you update Jest. That is the advantage of using imported types (`import {expect, test} from '@jest/globals'`) instead of globals which come from `@types/jest`.

For historical reasons, many users prefer globals instead of explicit imports. Typing the globals should be the main use of `@types/jest` at this moment. This use case is the reason why maintaining `@types/jest` makes sense.

To get rid of the lag there is an effort (#62037) to import types from `@jest/globals` into `@types/jest`. That is rather breaking change, so it must wait for a major release. It would be amazing to land this for Jest 30. Let’s see.
